itou/scripts/tests.py

```diff
@@ -152,3 +156,93 @@ def test_command_output(self):
+        )
+
+        # Suspension updated after the first script ran should not be updated.
+        suspension4 = SuspensionFactory(
+            approval__number="XXXXX0000003",
+            start_at=datetime.datetime.today() - relativedelta(days=30),
+            end_at=datetime.datetime.today() - relativedelta(days=1),
+        )
+        suspension5_updated_at = update_suspensions_end_at_mngt_comd.FIRST_SCRIPT_RUNNING_DATE + relativedelta(days=1)
+        suspension5 = SuspensionFactory(
+            approval=suspension4.approval, start_at=datetime.datetime.today(), updated_at=suspension5_updated_at
+        )
+
+        with tempfile.NamedTemporaryFile() as file:
+            suspensions = Suspension.objects.prefetch_related("approval").all()
+            for suspension in suspensions.iterator():
+                log_row = f"{suspension.approval.number} {suspension.start_at} {suspension.end_at}\n"
+                file.write(bytes(log_row, encoding="utf-8"))
+
+                # Approval who was deleted after the first script ran.
+                file.write(b"XXXXX5432764 2022-12-12 2024-12-11\n")
+            file.seek(0)
+
+            stdout = io.StringIO()
+            management.call_command("update_suspensions_end_at", file_path=file.name, wet_run=True, stdout=stdout)
+            output = stdout.getvalue()
+            # Ending after Suspension.MAX_DURATION_MONTHS
+            # suspension1 and suspension3
+            assert "Problematic suspensions found in database: 2." in output
+
+            # XXXXX0000001, XXXXX0000002, XXXXX0000003 and XXXXX5432764 (deleted approval).
+            assert "Total of unique approvals in input file: 4" in output
+
+            # suspension1 and suspension3
+            assert "Problematic suspensions found in input file: 2" in output
+
+            assert "Updated suspensions: 1" in output
+
+            # Edge cases
+            assert (
+                f"Skipping suspension updated after script ran: {suspension5.pk}, "
+                f"start_at: {suspension5.start_at.date()}, end_at: {suspension5.end_at.date()}"
+            ) in output
+            assert "Skipping approval not found: XXXXX5432764" in output
+            assert "Skipped suspensions (one suspension per approval): 1" in output
+            assert (
+                "Skipped suspensions (other reasons): 2" in output
+            )  # Approval not found, suspension updated after script ran.
+
+        suspension1.refresh_from_db()
+        # One suspension linked to one approval. Don't touch it.
+        assert not suspension1.updated_at
+
+        # Many suspensions linked to one approval: the end_at was updated many times.
+        # This is the first suspension. Only the last one should be updated.
+        suspension2.refresh_from_db()
+        assert not suspension2.updated_at
+        suspension3.refresh_from_db()
+        assert suspension3.updated_at
```

Comment:
Vu qu'il y a `freeze_gun` autant aller au bout de la logique, `updated_at` pourrais être initialisé lors de la création, surtout je doute que tout nos champs `updated_at` fonctionne de la même manière (corrigé un qui était jamais MAJ il n'y a pas longtemps :eyes:) :
```suggestion
        assert suspension3.updated_at == timezone.now()
```